generic_scripts/nlk.py

```diff
@@ -0,0 +1,304 @@
+
+        Parameters
+        ----------
+        y0 : float, optional
+            y coordinate of the horizontal plane.
+
+        Returns
+        -------
+        x_pos: numpy.array
+            Values of x where the field was computed.
+
+        fieldy: numpy.array
+            Vertical field in [T] at the plane y=y0.
+        """
+        x_pos = _np.linspace(-12, 12)[None, :]*1e-3
+        y_pos = _np.ones(x_pos.shape) * y0
+        r = _np.hstack([x_pos, y_pos])
+        fieldy = self.calc_magnetic_field(r)[1, :]
+        return x_pos[0], fieldy
+
+    def get_horizontal_magnetic_field(self, x0=0):
+        """NLK horizontal field at a vertical plane x=x0 for y ∈ [-12, 12] mm.
+
+        Parameters
+        ----------
+        x0 : float, optional
+            x coordinate of the horizontal plane.
+
+        Returns
+        -------
+        y_pos: numpy.array
+            Values of y where the field was computed.
+
+        fieldy: numpy.array
+            Horizontal field in [T] at the plane x=x0.
+        """
+        y_pos = _np.linspace(-12, 12)[None, :]*1e-3
+        x_pos = _np.ones(y_pos.shape) * x0
+        r = _np.hstack([x_pos, y_pos])
+        fieldy = self.calc_magnetic_field(r)[0, :]
+        return x_pos[1], fieldy
+
+    @staticmethod
+    def si_nlk_kick(
+            scale=1, fit_monomials=None, plot_flag=False,
+            r0=0.0):
+        """Generates the NLK polynom_b, its horizontal kick and the integrated
+        field at y=0. Useful for set the NLK pulse in the Sirius model.
+
+        Parameters
+        ----------
+        scale : float, optional
+            Multiplier of the NLK max field, by default 1.
+
+        fit_monomials : array like or list, optional
+            The multipoles indices or degrees of the terms included in the
+            polynom_b, by default is numpy.arange(10, dtype=int)
+
+        plot_flag : bool, optional
+            If True, plots the field profile, by default False.
+
+        r0 : float, optional
+            Closed orbit x coordinate, by default 0.0.
+
+        Returns
+        -------
+        x : numpy.array
+            Values of x where the field was computed.
+
+        integ_field : numpy.array
+            Integrated NLK field in [T.m].
+
+        kickx : numpy.array
+            Horizontal kick in [rad].
+
+        polynom_b : numpy array of shape (fit_monomials,)
+            Polynom B that describes the NLK kick.
+        """
+
+        nlk_length = 0.45
+        if fit_monomials is None:
+            fit_monomials = _np.arange(10, dtype=int)
+
+        nlk = NLK()
+        x, mag_field = nlk.get_vertical_magnetic_field()
+
+        brho, *_ = beam_rigidity(energy=3)  # energy in [GeV]
+        integ_field = scale * mag_field * nlk_length  # [T.m]
+        kickx = integ_field / brho  # [rad]
+
+        coeffs = _polyfit(x=x-r0, y=kickx, deg=fit_monomials)
+        fit_kickx = (x[:, None]**fit_monomials[None, :]) @ coeffs
```

Comment:
there is a method in `numpy.polynomial.polynomial` called `polyval`. I think you could used it here.

Reply:
I read the documentation and apparently `polyval` doesn't have the option to pass which monomials the powers of x were computed. I think this could introduce a bug in the case of `fit_monomials` isn't a consecutive sequence of numbers, i.e. `fit_monomials = [2,4,5,7,9]` or something like that.

Reply:
@jquentino take a look at this example. It works fine:
```
import numpy as np
import matplotlib.pyplot as mplt

monoms = np.array([2, 4, 5, 7, 9])
x = np.linspace(0, 2, 1001)

xp = x[::5]
yp = (xp[None, :]**monomials[:, None]).sum(axis=0)

coeffs = np.polynomial.polynomial.polyfit(xp, yp, deg=monoms)
yfit = np.polynomial.polynomial.polyval(x, coeffs)

print(coeffs)
print(coeffs[monoms])
mplt.plot(xp, yp, 'o')
mplt.plot(x, yfit)
mplt.show()
```

the point is that even though you define certain monomials to be fitted by the `polyfit` method, it always returns the coefficients for all the monomials up to the maximum you defined in the input, but it sets them to zero.
This way it is natural to use `polyval` to reconstruct the fitted curve afterwards.

With this in mind, it surprises me that your code runs without errors.

Reply:
Yes, you're right. I didn't pay attention to the fact that polyfit returns the coeffs with length equal to the maximum monomial +1, and with zeros in the monomials that weren't set. Thanks for the correction, I made the changes in the last commit.

Reply:
nice @jquentino !